flow/view_holder.cc

```diff
@@ -64,17 +63,21 @@ void ViewHolder::Create(zx_koid_t id,
   FML_DCHECK(bindings);
   FML_DCHECK(bindings->find(id) == bindings->end());
 
-  auto view_holder = std::make_unique<ViewHolder>(std::move(ui_task_runner),
-                                                  std::move(view_holder_token),
-                                                  on_bind_callback);
+  auto view_holder = std::make_unique<ViewHolder>(std::move(view_holder_token),
+                                                  std::move(on_view_created));
   bindings->emplace(id, std::move(view_holder));
 }
 
-void ViewHolder::Destroy(zx_koid_t id) {
+void ViewHolder::Destroy(zx_koid_t id, ViewIdCallback on_view_destroyed) {
   auto* bindings = tls_view_holder_bindings.get();
   FML_DCHECK(bindings);
 
-  bindings->erase(id);
+  auto binding = bindings->find(id);
+  FML_DCHECK(binding != bindings->end());
+
+  if (binding->second->view_holder_) {
+    on_view_destroyed(binding->second->view_holder_->id());
+  }
```

Comment:
It looks like the ViewHolder object is left in bindings. Is the object itself destroyed elsewhere?

Reply:
Ooops!  also good catch :+1: